### PR TITLE
I18n cleanup!

### PR DESCRIPTION
**Pull request by viroulep**
While writing the wiki about how to translate the website I realized we may not actually need the applications-specific locales in our `config` directory.

Turned out I removed all of them but `carrierwave` and `time_will_tell`.
- The first one because they changed `errors.messages.extension_white_list_error` to `errors.messages.extension_whitelist_error`: there is no version of `carrierwave-i18n` compatible with our version of `carrierwave`, and updating the latter is not possible because we would need to update `carrierwave-crop` for which we use a specific version from @jfly.
- The second one because no `time_will_tell-i18n` is available.

For the other gems I added `gemname-i18n` to the Gemfile, I also looked at the history for these file to spot the key we added/overrode, and added them to our main locale files.
(For `devise-bootstrap-views` I just updated the version since they added a lot of locales later on)

The `devise.failure.{invalid,not_found_in_database}` keys were a pain because I couldn't prefix them with `wca.`, so I had to tweak the `i18n-tasks` config file to be able to check they are present/used!
(FYI I tried playing with [`DeviseController#set_flash_message`](https://github.com/plataformatec/devise/blob/8a4d610c587ca7d0f5ce40d9a4a28a2da717f1e9/app/controllers/devise_controller.rb#L152) by creating our own controller for sessions, but it wouldn't work for these specific messages because `set_flash_message` is actually not used in the base [`session_controller#new`](https://github.com/plataformatec/devise/blob/8a4d610c587ca7d0f5ce40d9a4a28a2da717f1e9/app/controllers/devise/sessions_controller.rb#L8-L13))

There is at least two main benefits for these changes:
- our locales for external gems will stay up to date (our doorkeeper locale was very outdated)
- these external gems have translations for most languages already and provide them with no work from our side :)

### CODE REVIEW

File: WcaOnRails/spec/requests/users_spec.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
     post_via_redirect user_confirmation_path, 'user[email]' => user.email
     expect(response).to be_success
-    expect(response.body).to include("You will receive an email with instructions about how to confirm your account in a few minutes.")
+    expect(response.body).to include(I18n.t('devise.confirmations.send_instructions'))
```
Comment (viroulep) [inline WcaOnRails/spec/requests/users_spec.rb]: For the record the new text is:
> You will receive an email with instructions for how to confirm your email address in a few minutes.

File: WcaOnRails/app/views/oauth/applications/_form.html.erb [hunk 2]
```diff
@@ -33,7 +33,7 @@
       <%= f.text_field :scopes, class: 'form-control' %>
       <%= doorkeeper_errors_for application, :scopes %>
       <span class="help-block">
-        <%= raw t('doorkeeper.applications.help.scopes') %>
+        <%= raw t('wca.doorkeeper.applications.help.scopes') %>
```
Comment (jfly) [inline WcaOnRails/app/views/oauth/applications/_form.html.erb]: Oh, possibly better solution that using `raw`... can we change this to `t('wca.doorkeeper.applications.help.scopes_html')` instead?

File: WcaOnRails/config/locales/en.yml [hunk 3]
```diff
@@ -335,6 +335,11 @@ en:
       wrong_size: "is the wrong size (should be %{file_size})"
       size_too_small: "is too small (should be at least %{file_size})"
       size_too_big: "is too big (should be at most %{file_size})"
+  #context: Overriden key for sign in
```
Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: Nicely caught! How did you find these?

File: WcaOnRails/config/i18n-tasks.yml.erb [hunk 4]
```diff
@@ -85,7 +85,11 @@ search:
 #   api_key: "AbC-dEf5"
 
 ## Do not consider these keys missing:
-# ignore_missing:
+ignore_missing:
+  # Kaminari's keys, they are not prefixed by "kaminari.".
+  - 'views.pagination.*'
+  # We override devise.failure.(...) without adding the "wca" scope, but we can safely ignore the others
+  - 'devise.{mailer,passwords,registrations,sessions,shared,confirmations,unlocks}.*'
```
Comment (jfly) [inline WcaOnRails/config/i18n-tasks.yml.erb]: Dang, this looks tricky =( What would happen if we specified `devise.*` instead?

**Comment by jfly** [comment]
> and updating the latter is not possible because we would need to update carrierwave-crop for which we use a specific version from @jfly.

Urg, sorry! The profile picture stuff is a total mess, and needs to be rethought. I'll take care of that someday.

File: WcaOnRails/config/i18n-tasks.yml.erb (hunk 4 above)
Comment (viroulep) [inline WcaOnRails/config/i18n-tasks.yml.erb]: Our `devise.failure` overridden keys would not be detected as missing if they were actually missing in an available language.

Comment (jfly) [inline WcaOnRails/config/i18n-tasks.yml.erb]: Ah, right. For some reason, I was thinking that other languages were compareed again our english translations to see if anything is missing, but instead, they are compared against our codebase.

**Comment by viroulep** [comment]
I plan to merge this on Sunday, unless someone requests changes!

**Pull request merged**